File: 0345-reverse-vowels-of-a-string/0345-reverse-vowels-of-a-string.py
class Solution:
    def reverseVowels(self, s: str) -> str:
        
        vowelSet = set("aeiouAEIOU")
        vowelString = ""
        
        for item in s:
            if item in vowelSet:
                vowelString += item
        
        vowelString = vowelString[::-1]
        
        res = ""
        i = 0
        
        for item in s:
            if item in vowelSet:
                res += vowelString[i]
                i+=1
            else:
                res += item
        return res
                

File: 0345-reverse-vowels-of-a-string/0345-reverse-vowels-of-a-string.cpp
class Solution {
public:
    string reverseVowels(string s) 
    {
        set<char>vowels = {'a','e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U'};
        
        int size = s.size();
        
        string vowelString = "";
        
        for (int i = 0; i < size; i++){
            
            if (vowels.find(s[i]) != vowels.end()) 
                vowelString += s[i];
        }
        
        int j = vowelString.size()-1;
        
        for (int i = 0; i < size; i++){
            if (vowels.find(s[i]) != vowels.end()){
                s[i] = vowelString[j];
                j-=1;
            }  
        }
        return s;
    }
};